pyccl/halos/profiles.py

```diff
@@ -871,15 +874,14 @@ def _get_cM(self, cosmo, M, a, mdef=None):
         return self.cM.get_concentration(cosmo, M, a, mdef_other=mdef)
 
     def _get_alpha(self, cosmo, M, a, mdef):
-        if self.alpha is None:
+        if self.alpha == 'cosmo':
             mdef_vir = MassDef('vir', 'matter')
             Mvir = mdef.translate_mass(cosmo, M, a, mdef_vir)
-            print(cosmo, M, a, mdef_vir)
             sM = sigmaM(cosmo, Mvir, a)
             nu = 1.686 / sM
             alpha = 0.155 + 0.0095 * nu * nu
         else:
-            alpha = self.alpha
+            alpha = np.array([self.alpha])
```

Comment:
This will give alpha an extra array dimension no matter what the dimensionality of alpha is. I think what we actually want here is `np.atleast_1d`.

Reply:
Fixed

Reply:
Just found out alpha should have the same shape as `M`.

Reply:
How about allowing the possibility for `alpha` to be a function?

Reply:
Yes, assuming that's what you guys need?

Reply:
Sorry, I'm a little bit confused. Right now `_get_alpha` is a function of `cosmo`, `M`, `a` and `mdef`. If we want to calculate alpha from cosmology, then all four variables will be used. But if we want to set alpha to our own value, then only the shape of `M` is used. Do you mean defining a new function `def alpha`... to deal with the two cases?

Reply:
Is there a better way to retrieve alpha when alpha is a free parameter? without first going through `_get_alpha`, which requires unused arguments (`cosmo`, `a` and `mdef`). What I can think of is a `if` before `norm = self._norm(M_use, R_s, c_M, alpha)
` to separate the two cases and it returns `alpha = np.array([self.alpha])` directly.

Reply:
These are used arguments are in a private function so it doesn't really matter. Either way, `cosmo`, `a`, and `mdef` are always in the scope of `_real`. You could compress the if-statement if you take the special case outside, but then the code wouldn't be optimized.

If it always has to be the same shape as `M`, why not cast it to `M` using `broadcast_to` inside of the `_get_alpha` function? I think all of the `alpha` treatment should be in that function to properly enclose the calculation.

Then, you could add one more clause to the statement checking whether `alpha` is a function of mass, and output whatever `alpha` needs to be used downstream.

Reply:
Great, I see. I just wanted to make sure it was not a bad practice to ignore some of the arguments. I am using `alpha = np.full_like(M, self.alpha)` and it is a little bit faster than `np.broadcast_to(self.alpha, M.shape)`.